drivers/scheduler/k8s/k8s.go

```diff
@@ -616,6 +616,72 @@ func (k *k8s) GetNodesForApp(ctx *scheduler.Context) ([]node.Node, error) {
 	return result, nil
 }
 
+func (k *k8s) GenerateSupportBundle(ctx *scheduler.Context) (string, error) {
+	var output string
+	var err error
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			output += insertLineBreak(obj.Name)
+			var depStatus apps_api.DeploymentStatus
+			if depStatus, err = k8sutils.DescribeDeployment(obj); err != nil {
+				output += fmt.Sprintf("%v", &ErrFailedToValidateApp{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get status of deployment: %v. Err: %v", obj.Name, err),
+				})
+			}
+			//Dump depStatus
+			output += fmt.Sprintf("%v\n", depStatus)
+			pods, _ := k8sutils.GetDeploymentPods(obj)
+			for _, pod := range pods {
```

Comment:
Create common function for generating output from a list of pods. This block and line 658..667 have common code.